src/views/ViewAllExplorer.tsx

```diff
@@ -16,20 +16,38 @@ import { WcConnectionCtrl } from '../controllers/WcConnectionCtrl';
 import type { RouterProps } from '../types/routerTypes';
 import useTheme from '../hooks/useTheme';
 import { ThemeCtrl } from '../controllers/ThemeCtrl';
+import { UiUtil } from '../utils/UiUtil';
 
 function ViewAllExplorer({
   isPortrait,
   windowHeight,
   windowWidth,
 }: RouterProps) {
   const Theme = useTheme();
-  const optionsState = useSnapshot(OptionsCtrl.state);
-  const wcConnectionState = useSnapshot(WcConnectionCtrl.state);
-  const themeState = useSnapshot(ThemeCtrl.state);
-  const loading = !optionsState.isDataLoaded || !wcConnectionState.pairingUri;
-  const wallets = useMemo(() => {
-    return ExplorerCtrl.state.wallets.listings;
-  }, []);
+  const { isDataLoaded } = useSnapshot(OptionsCtrl.state);
+  const { pairingUri } = useSnapshot(WcConnectionCtrl.state);
+  const { themeMode } = useSnapshot(ThemeCtrl.state);
+  const { wallets, recommendedWallets } = useSnapshot(ExplorerCtrl.state);
+  const [walletsLoading, setWalletsLoading] = useState(false);
+  const loading = !isDataLoaded || !pairingUri || walletsLoading;
+  const _wallets = [...recommendedWallets, ...wallets.listings];
+
+  useEffect(() => {
+    if (!loading) {
+      UiUtil.layoutAnimation();
+    }
+  }, [loading]);
+
+  useEffect(() => {
+    async function getWallets() {
+      if (wallets.listings.length === 0) {
+        setWalletsLoading(true);
+        await ExplorerCtrl.getWallets();
+        setWalletsLoading(false);
+      }
+    }
+    getWallets();
+  }, [wallets]);
```

Comment:
This is a bit sketchy for hooks (passing array or object). Since you rely solely on length of walelts, you could create separate var `walletLength` and use that inside the hook and as it's dependency